### ISSUE TITLE
Misleading implementation of `I_stim`

### ISSUE THREAD

**Issue by Silmathoron** [nest/nestml#449]
If I understand the model files correctly, the input current from e.g. `step_current_generator` in NEST is currently passed to the `I_stim` variable

    # Input current injected by CurrentEvent.
    # This variable is used to transport the current applied into the
    # _dynamics function computing the derivative of the state vector.
    I_stim pA = 0pA

However, I believed that the injected current would actually be the ``curents`` variable defined in the ``input`` block

    input:
      spikesInh nS  <- inhibitory spike
      spikesExc nS  <- excitatory spike
      currents <- current
    end

Yet, I also noticed that this has no unit defined, which seems surprising...

So, in a nutshell, which is which and how do they differ?

**Comment by Silmathoron** [comment]
So following the later issue #450 on current missing physical units, I guess we agree that ``currents`` need one.
From this I would guess that it makes ``I_stim`` irrelevant and therefore that this variable should disappear.
Any thought @clinssen @jougs ?

**Comment by jougs** [comment]
@Silmathoron: sorry, I completely missed this one and only got drawn to it through #526. Your initial assessment was totally correct and #526 will fix all your points. Yay!
